### PR TITLE
[Fix #7737] Add new `Style/RedundantArguments` cop

### LINKED ISSUE #7737: Remove unnecessary arguments for builtin methods.
## Is your feature request related to a problem? Please describe.

I found something like `array.join('')` in my code recently. This can of course be replaced by `array.join`. There is also `string.split("\n")` or `string.gsub(a, '')`. I will try to collect more cases like this in this issue (help is welcome).

## Describe the solution you'd like

For a list of important builtin functions (e.g. `String#split` and `Array#join`), if an argument is passed explicitly that leads to the same behavior as leaving the argument away, suggest to remove that argument.

## Additional context

Note that I might do this myself eventually. Depends on how much time I have and how eager someone else is to take it.

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes #7737

Previous PR https://github.com/rubocop-hq/rubocop/pull/7761

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop-hq/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -3977,6 +3977,15 @@ Style/RandomWithOffset:
   Enabled: true
   VersionAdded: '0.52'
 
+Style/RedundantArguments:
+  Description: 'Check for redundant arguments to methods.'
+  Enabled: pending
+  Safe: false
+  VersionAdded: '1.4'
```
Comment (bbatsov) [inline config/default.yml:3984]: Now we use `<<next>>` here.

File: lib/rubocop/cop/style/redundant_arguments.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    module Style
+      # This cop checks for redundant arguments to methods.
+      # Method names & arguments can be configured like:
+      #
+      # RedundantArguments:
+      #   join: ''
+      #   split: ' '
+      #   foo: 2
+      #
+      # Limitations:
```
Comment (bbatsov) [inline lib/rubocop/cop/style/redundant_arguments.rb:16]: I'd put those limitations before the configuration examples.

File: lib/rubocop/cop/style/redundant_arguments.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      # 1. This cop matches for method names only and hence cannot tell apart
+      #    methods with same name in different classes.
+      # 2. This cop is limited to methods with single parameter.
+      # 3. This cop is unsafe if certain kernel global variables are set.
+      #
+      # @example
+      #   # bad
+      #
```
Comment (bbatsov) [inline lib/rubocop/cop/style/redundant_arguments.rb:24]: This blank line is redundant.

File: lib/rubocop/cop/style/redundant_arguments.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      #   A.foo(2)
+      #
+      #   # good
+      #   array.join
+      #   [1, 2, 3].join
+      #   string.split
+      #   "first second".split
+      #   A.foo
+      class RedundantArguments < Cop
+        MSG = 'Argument is redundant.'
```
Comment (bbatsov) [inline lib/rubocop/cop/style/redundant_arguments.rb:38]: I'd expand the message to mention the argument in question and some text saying that the argument is implied by default, so it's clearer what makes it redundant.

**Comment by tejasbubane** [comment]
@bbatsov changes done

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**